src/settings/Settings.tsx

```diff
@@ -532,6 +532,14 @@ export const SETTINGS: { [setting: string]: ISetting } = {
         labsGroup: LabGroup.Rooms,
         default: false,
     },
+    // MSC3952 intentional mentions support.
+    "feature_intentional_mentions": {
```

Comment:
Does this not require server MSC support, might want to use ServerSupportUnstableFeatureController

Reply:
For encrypted rooms it essentially doesn't require server support since:

* (For mobile) the server pushes all encrypted messages to devices.
* The event fields this operates on are encrypted.

The answer for unencrypted rooms is probably a bit more complicated. I think it is possible for the server & client to disagree somewhat about whether an event is a highlight or not (e.g. if the server matches on `.m.rule.contains_display_name` while the client matches on `.org.matrix.msc3952.is_user_mention`).

I'm not 100% sure of the impact of this, to be honest, I vaguely think the counts in the room might get a little wonky, but I think >95% of the time it will be fine.

Reply:
The impact needs figuring out for the MSC surely, given that clients depend on the server to calculate for unencrypted rooms and aggregate in the /sync notification counts

Reply:
There's a section devoted to backwards compatibility in the MSC: https://github.com/matrix-org/matrix-spec-proposals/blob/clokep/intentional-mentions/proposals/3952-intentional-mentions.md#backwards-compatibility.

Reply:
From a brief skim I'm not seeing anything there about mismatched support between client & server.

> To avoid unintentional mentions these rules are modified to only apply when the m.mentions property is missing; clients should provide at least an empty m.mentions property on every message to avoid the unintentional mentions discussed above.

If a client is respecting this but a server isn't then it seems like a room may end up with a red highlight count (server generated) but the event not highlighted due to implementation mismatches

Reply:
The last paragraph talks about it (it is pretty much the same as what I described above though):

> After acceptance, it is likely for there to be disagreement about which push rules are implemented: legacy clients and homeservers may not yet have deprecated the `.m.rule.contains_display_name`, `.m.rule.contains_user_name`, and `.m.rule.roomnotif push` rules, while up-to-date clients and homeservers will support the `.m.rule.is_user_mention` and `.m.rule.is_room_mention` push rules. It is expected that both sets of push rules will need to be supported for a period of time, but at worst case should simply result in the current behavior (documented in the preamble).

> If a client is respecting this but a server isn't then it seems like a room may end up with a red highlight count (server generated) but the event not highlighted due to implementation mismatches

Yes, that seems like the most likely impact. I think this is most likely to be rare though since it is expected that usually both rules would match the same events. (This isn't true if someone is sending events maliciously in some way, however, but I think in normal messages it is unlikely to happen.)

Reply:
Surely that'd be mitigated if client's pivoted their support based on server's capabilities?

Reply:
I don't have too strong of opinions -- the MSC hasn't seen much input from client developers. It sounds like you think it should be keyed on it though, so I can add that.

Right now the MSC suggests enabling the feature if it sees one of the push rules come down. It looks like `ServerSupportUnstableFeatureController` really wants to use a feature flag though, so I can update the MSC to that.